src/api_midlevel.jl

```diff
@@ -46,6 +46,83 @@ function get_extent_dims(obj::Union{Dataspace,Dataset,Attribute})
     return dims, maxdims
 end
 
+
+"""
+    HDF5.get_chunk_offset(dataset_id, index)
+
+Get 0-based offset of chunk from 0-based index.
+
+For a 1-based API, see HDF5.ChunkStorage.
+
+The returned offsets are in Julian order than HDF5's C-order.
+"""
+function get_chunk_offset(dataset_id, index)
+    extent = size(dataset_id)
+
+    chunk = get_chunk(dataset_id)
+    chunk_indices = CartesianIndices( ntuple(i->0:extent[i]÷chunk[i]-1, length(extent)) )
+    offset = hsize_t.(chunk_indices[index + 1].I .* chunk)
+
+    offset
+end
+
+"""
+    HDF5.get_chunk_index(dataset_id, offset)
+
+Get 0-based index of chunk from 0-based offset in Julian order.
+
+For a 1-based API, see HDF5.ChunkStorage.
+
+This should be equivalent to h5d_get_chunk_info_by_coord( dataset_id, reverse(offset) )[:index].

```

Comment:
h5d_get_chunk_info_by_coord did you mean  h5d_get_chunk_info?

Reply:
I believe this is actually `h5d_get_chunk_info_by_coord` which takes a multidimensional coordinate as an input and returns the following.

1. `filter_mask`
2. `addr`, the chunk address (0-based linear index)
3. `size`, the chunk size in bytes

In this case, we want the chunk address. Rather than `:index` the keyword should be `:addr`.
https://portal.hdfgroup.org/display/HDF5/H5D_GET_CHUNK_INFO_BY_COORD

The new function `get_chunk_offset` is equivalent to `h5d_get_chunk_info` which takes a 0-based linear index, and returns a 0-based multidimensional offset.

Reply:
Hmm... I am definitely a bit confused here myself. I'll collect my thoughts and write some tests.